p2p/transport/webrtc/udpmux/packetqueue.go

```diff
@@ -0,0 +1,124 @@
+package udpmux
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+
+	pool "github.com/libp2p/go-buffer-pool"
+)
+
+const maxPacketsInQueue int = 128
+
+type packet struct {
+	addr net.Addr
+	buf  []byte
+}
+
+var (
+	errTooManyPackets = fmt.Errorf("too many packets in queue; dropping")
+)
+
+// packetQueue is a blocking queue for buffering packets received
+// on net.PacketConn instances. Packets can be pushed by using
+// `push` and read by using `pop`
+type packetQueue struct {
+	ctx         context.Context
+	cancel      context.CancelFunc
+	mu          sync.Mutex
```

Comment:
Your use of mutexes are pretty low-level here, you may want to wrap it in some type as to ensure it's always properly locked and unlocked!